File: Strings/StringKeySort.cpp
#include<bits/stdc++.h>
using namespace std;

int col = 0;
vector<string> splitString(string str) {
    stringstream ss(str);
    string token;
    vector<string> tokens;
    while(getline(ss, token, ' ')) {
        tokens.push_back(token);
    }
    
    return tokens;
}

vector<vector<string>> tokenise(vector<string> & v) {
    int n = v.size();
    vector<vector<string>> op;
    for(int i=0; i<n; i++) {
        string cs = v[i];
        vector<string> tokens = splitString(cs);
        op.push_back(tokens);
    }
    
    return op;
}

bool compare(vector<string> & v1, vector<string> & v2) {
    return stoi(v1[col-1]) < stoi(v2[col-1]);
}

void stringKeySort(vector<string> & v, int c, bool reverse, string type) {
    vector<vector<string>> tokens = tokenise(v);
    col = c;
    sort(tokens.begin(), tokens.end(), compare);
    
    for(vector<string> v : tokens) {
        for(string s : v) {
            cout<<s<<" ";
        }
        cout<<endl;
    }
}

int main() {
    int n;
    cin>>n;
    vector<string> v = {"92 022", "82 12", "77 13"};
    
    int c = 2;
    bool reverse = false;
    string type = "numeric";
    
    stringKeySort(v, c, reverse, type);
}

File: Strings/StringKeySort.js
let col;

function compareNumeric(arr1, arr2) {
  return Number(arr1[col]) - Number(arr2[col]);
}

function compareString(arr1, arr2) {
  if (arr1[col] < arr2[col]) return -1;
  else if (arr1[col] === arr2[col]) return 0;
  return 1;
}

function StringKeySort(arr, c, type) {
  const string_arr = arr.map((el) => el.split(" "));
  col = c - 1;
  if (type === "numeric") {
    string_arr.sort(compareNumeric);
  } else {
    string_arr.sort(compareString);
  }

  console.log(string_arr);
}

const arr = ["92 022", "82 12", "77 13"];

const res = StringKeySort(arr, 2, "numeric");
